src/include/main/database.h

```diff
@@ -95,12 +95,15 @@ class Database {
      *        an in-memory database.
      * @param systemConfig System configurations (buffer pool size and max num threads).
      */
-    KUZU_API explicit Database(std::string_view databasePath,
+    KUZU_API static std::unique_ptr<Database> construct(std::string_view databasePath,
         SystemConfig systemConfig = SystemConfig());
+    KUZU_API static std::shared_ptr<Database> constructShared(std::string_view databasePath,
```

Comment:
`std::shared_ptr` has a constructor that takes a `std::unique_ptr`, so I'm not sure if we really need a second constructor just to produce a shared_ptr.

I also don't understand the purpose of separating out `initMembers` or making the constructor private in favour of a static `construct` function. They don't seem like they do anything that couldn't be achieved with a regular constructor.

Reply:
Thanks pointing that out, I'll remove the second constructor.

I separated `initMembers` out of the constructor because the vtable for a class isn't initialized until after the constructor for the base class completes execution. I wanted to have `initBufferManager` be overrideable for tests but this is only possible outside of the constructor for `Database`